src/controller/logging.cpp

```diff
@@ -85,14 +83,18 @@ inline QString removeAbsolutePathPrefix(const QString& filePath)
 
 void messageHandler(QtMsgType type, const QMessageLogContext& context, const QString& message)
 {
+    std::cerr << toString(type) << ": " << message.toStdString() << std::endl;
```

Comment:
I think the mutex lock on lines 93-94 should be moved up before this line, otherwise two simultaneous threads writing to stderr may garble it.